MPF.Modules/BaseParameters.cs

```diff
@@ -1849,10 +1849,33 @@ protected static string GetPlayStation5Version(char? driveLetter)
                     case 'A': return Region.Asia;
                     case 'C': return Region.China;
                     case 'E': return Region.Europe;
-                    case 'J': return Region.JapanKorea;
                     case 'K': return Region.SouthKorea;
-                    case 'P': return Region.Japan;
                     case 'U': return Region.UnitedStatesOfAmerica;
+                    case 'P': 
+                        // Region of S_P_ serials may be Japan, Asia, or SouthKorea
+                        switch (serial[3])
+                        {
+                            case 'S':
+                                // Check first two digits of S_PS serial
+                                switch (serial.Substring(4, 2))
+                                {
+                                    case "46": return Region.SouthKorea;
+                                    case "56": return Region.SouthKorea;
+                                    case "51": return Region.Asia;
+                                    case "55": return Region.Asia;
+                                    default: return Region.Japan;
+                                }
+                            case 'M':
+                                // Check first three digits of S_PM serial
+                                switch (serial.Substring(4, 3))
+                                {
+                                    case "645": return Region.SouthKorea;
+                                    case "675": return Region.SouthKorea;
+                                    case "885": return Region.SouthKorea;
+                                    default: return Region.Japan; // Remaining S_PM serials may be Japan or Asia
+                                }
```

Comment:
This feels really difficult to keep up to date, I'll be honest. I understand that it's far more accurate, though.